src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -131,29 +131,41 @@ export default class SquarePaymentStrategy implements PaymentStrategy {
             ...this._paymentMethod?.initializationData,
             callbacks: {
                 cardNonceResponseReceived: (errors, nonce, cardData, billingContact, shippingContact) => {
-                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
-                        this._handleWalletNonceResponse(errors, nonce, cardData, billingContact, shippingContact);
+                    const deferredRequest = this._getDeferredRequestNonce();
+                    const { onError = noop } = this._getInitializeOptions();
+
+                    if (!nonce) {
+                        onError(errors);
+
+                        return deferredRequest.reject(get(errors, '0', {}));
+                    }
+
+                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none && nonce) {
+                        this._handleWalletNonceResponse(nonce, cardData, billingContact, shippingContact);
                     } else {
                         if (this._is3DSExperimentOn()) {
                             this._getPaymentForm().verifyBuyer(
                                 nonce,
                                 this._getVerificationDetails(),
                                 (error: SquareVerificationError, verificationResults: SquareVerificationResult) => {
+
                                     if (!isEmpty(error)) {
-                                        return this._getDeferredRequestNonce().reject(error);
+                                        onError(error);
+
+                                        return deferredRequest.reject(get(error, '0', {}));
                                     }
 
-                                    this._handleCardNonceResponse(errors, nonce, verificationResults.token);
+                                    deferredRequest.resolve({ nonce: JSON.stringify({ nonce, token: verificationResults.token }) });
```

Comment:
The line below https://github.com/bigcommerce/checkout-sdk-js/pull/1247/files#diff-5887d823b45e38ded51b5b326abf6d0ec989b5ca2f2ab04d635009918d6a397aL206 used to just resolve nonce and token, and here we are stringifying them. Does that cause other issues?

Reply:
@animesh1987 in bigpay we only receive `credit_card_token` and thats because the `nonce` is converted to credit_card_token in bigpay-client-js so the `token` we are sending here is ignored, so now we  parse this json in this bigpay [PR](https://github.com/bigcommerce/bigpay/pull/4341/files#diff-ae35a26aa3f24862fdcf5cb2adcc0c864c3b19890dc6cd8ece985e877879b653R309)